From http://fivethirtyeight.com/features/can-you-slay-the-puzzle-of-the-monsters-gems/

A video game requires you to slay monsters to collect gems. Every time you slay a monster, it drops one of three types of gems: a common gem, an uncommon gem or a rare gem. The probabilities of these gems being dropped are in the ratio of 3:2:1 — three common gems for every two uncommon gems for every one rare gem, on average. If you slay monsters until you have at least one of each of the three types of gems, __how many of the most common gems will you end up with, on average?__

In [1]:
import numpy as np
from random import choice
from collections import defaultdict

In [2]:
GEM_COMMON=1
GEM_UNCOMMON=2
GEM_RARE=3
GEM_PROB_LIST=[GEM_RARE,
               GEM_UNCOMMON,GEM_UNCOMMON,
               GEM_COMMON,GEM_COMMON,GEM_COMMON]
def slay():
    return choice(GEM_PROB_LIST)

def slay_until_done():
    gem_count = defaultdict(int)
    gem_set = set()
    done_set = set()
    done_set.add(GEM_COMMON)
    done_set.add(GEM_UNCOMMON)
    done_set.add(GEM_RARE)
    while gem_set != done_set:
        gem = slay()
        gem_count[gem] += 1
        gem_set.add(gem)
    return gem_count

def run_get_common_list(N):
    common_list = np.zeros(N)
    for i in range(N):
        common_list[i] = slay_until_done()[GEM_COMMON]
    return common_list

def answer(N):
    print("for {} game runs, the average common count is {}".format(N,np.average(run_get_common_list(N))))
    

In [3]:
answer(10)
answer(100)
answer(1000)
answer(10000)
answer(100000)
answer(1000000)

for 10 game runs, the average common count is 3.3
for 100 game runs, the average common count is 3.73
for 1000 game runs, the average common count is 3.687
for 10000 game runs, the average common count is 3.6404
for 100000 game runs, the average common count is 3.65996
for 1000000 game runs, the average common count is 3.648137


So, we get about 3.65 common gems per game.  That rounds to 4 gems if we're talking about whole gems.